scl/darwinosl/plugin.conf

```diff
@@ -0,0 +1,79 @@
+#############################################################################
+# Copyright (c) 2023 Hofi <hofione@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+# NOTE: For predicate hints see:
+#						log help predicates
+@define def-osl-predicate "    (eventType == 'logEvent' || eventType == 'lossEvent' || eventType == 'stateEvent' || eventType == 'userActionEvent')
+													  && (logType != 'debug')
+													"
+
+block source darwin-oslog(...) {
+@requires darwinosl "The darwinosl() source driver depends on the darwinosl module"
+	darwinosl(
+		filter-predicate("`def-osl-predicate`")
+		`__VARARGS__`
+		flags(syslog-protocol dont-store-legacy-msghdr assume-utf8)
+	);
+};
+
+block source darwin-oslog-stream(params("--style ndjson --type log --type trace --level info --level debug") user-params("")) {
+	channel {
+		source {
+			program(
+				"/usr/bin/log stream `params` `user-params`"
+				flags(
+					no-parse
+					no-multi-line
+					dont-store-legacy-msghdr
+					assume-utf8
+				)
+			);
+		};
+
+		parser {
+	    json-parser ();
+			date-parser(
+				format("%Y-%m-%d %T.%f%z")
+				template("$timestamp")
+			);
+	    add-contextual-data(
+				selector("$messageType"),
+				database("../share/syslog-ng/include/scl/darwinosl/darwinosl-metadata-db.csv"),
+				default-selector("unknown"),
+				prefix(".priorities.")
+			);
+		};
+
+		rewrite {
+			subst(
+					" ",
+					"\\0x20",
```

Comment:
I think we should use underscore here, we should only change $PROGRAM and not the processImagePath field.

Reply:
In the first version I also wanted to use underscore, but escaping it could signal for any later possible parsing attempt what the original value was.
Does it make sense?